### PR TITLE
Feature/tvos support

### PR DESCRIPTION
**Pull request by vimanyu**
This is specifically for building tvOS libraries for both device and simulator. For testing the built libraries, there will be another PR that ensures all integration tests are working.

- single toolchain file that could possibly be used in the future for iOS as well as mac builds. Currently being used only for tvOS.
- Disabled AdMob and Dynamic Links as they are not supported on tvOS.
- Disabled some functionality from auth (PhoneAuthProvider) that doesn't exist on tvOS.
- Disabled some functionality in FCM (since some of the frameworks we are using do not exist on tvOS). FCM might require more work for full functionality on tvOS but it doesn't block the builds. 

Easiest way to build for tvos (including xcframeworks) is to use the helper bash scripts,
`build_scripts/tvos/build.sh`

### CODE REVIEW

File: build_scripts/tvos/install_prereqs.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
+#!/bin/bash -e
+
+if [[ $(uname) != "Darwin" ]]; then
+    echo "Unsupported platform, iOS can only be build on a MacOS machine."
```
Comment (DellaBitta) [inline build_scripts/tvos/install_prereqs.sh:4]: tvOS, not iOS

File: build_scripts/tvos/package.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#!/bin/bash -e
+
```
Comment (DellaBitta) [inline build_scripts/tvos/package.sh:2]: This should have some sort of header, like our other package scripts.

File: cmake/toolchains/apple.toolchain.cmake [hunk 3]
```diff
@@ -0,0 +1,926 @@
+# This file is part of the ios-cmake project. It was retrieved from
```
Comment (DellaBitta) [inline cmake/toolchains/apple.toolchain.cmake:1]: I think you should add the Google copyright to the top of this file, too, and also create a entry for this in the `release_build_files/NOTICES` file.

**Review by DellaBitta — CHANGES_REQUESTED** [review]

File: app/src/app_common.cc [hunk 4]
```diff
@@ -88,7 +88,7 @@ const char* kCpuArchitecture = "arm32";
 
 #elif defined(__APPLE__)
 const char* kCppRuntimeOrStl = "libcpp";
-#if FIREBASE_PLATFORM_IOS
+#if FIREBASE_PLATFORM_IOS || FIREBASE_PLATFORM_TVOS
 const char* kOperatingSystem = "ios";
```
[the remark this answers is not in the document]
Comment (vimanyu) [reply app/src/app_common.cc:92]: Great catch and thank you for spotting this!

**Comment by vimanyu** [comment]
Thanks for the comments! All have been addressed!

File: build_scripts/tvos/package.sh [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    echo "Usage: $0 <built tvOS SDK path> <path to put packaged files into>
+example:
+  build_scripts/tvos/package.sh tvos_build package_out"
+}
+
+builtpath=$1
+packagepath=$2
+
+if [[ -z "${builtpath}" || -z "${packagepath}" ]]; then
+    echo "Usage: $0 <built tvOS SDK path> <path to put packaged files into>"
```
Comment (DellaBitta) [inline build_scripts/tvos/package.sh:16]: call usage() here intead.

File: build_scripts/tvos/package.sh [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    exit 1
+fi
+
+if [[ ! -d "${builtpath}/xcframeworks" ]]; then
+    echo "Built tvOS SDK not found at path '${builtpath}'."
```
Comment (DellaBitta) [inline build_scripts/tvos/package.sh:21]: Call usage() here, too.

**Review by DellaBitta — CHANGES_REQUESTED** [review]

File: build_scripts/tvos/package.sh (hunk 5 above)
Comment (vimanyu) [reply build_scripts/tvos/package.sh]: oops! Missed it! Done!

**Review by DellaBitta — APPROVED** [review]

**Pull request merged**